ddl/sequence_test.go

```diff
@@ -660,6 +660,19 @@ func (s *testSequenceSuite) TestSequenceFunction(c *C) {
 	s.tk.MustQuery("select setval(seq, -10)").Check(testkit.Rows("-10"))
 	s.tk.MustQuery("select setval(seq, -5)").Check(testkit.Rows("<nil>"))
 	s.tk.MustExec("drop sequence seq")
+
+	// test the current value already satisfied setval in other session.
+	s.tk.MustExec("create sequence seq")
+	s.tk.MustQuery("select setval(seq, 100)").Check(testkit.Rows("100"))
+	se, err := session.CreateSession4Test(s.store)
+	c.Assert(err, IsNil)
+	tk1 := testkit.NewTestKit(c, s.store)
+	tk1.Se = se
+	tk1.MustExec("use test")
+	tk1.MustQuery("select setval(seq, 50)").Check(testkit.Rows("<nil>"))
+	tk1.MustQuery("select setval(seq, 100)").Check(testkit.Rows("<nil>"))
+	tk1.MustQuery("select setval(seq, 101)").Check(testkit.Rows("101"))
```

Comment:
Also check the real sequence value by selecting `nextval()`.
And check the situation when increment < 0.

Reply:
ok

Reply:
Addressed @djshow832